charmcraft/linters.py

```diff
@@ -233,10 +235,86 @@ def run(self, basedir: pathlib.Path) -> str:
         return result
 
 
+class JujuActions:
+    """Check that the actions.yaml file is valid YAML if it exists."""
+
+    check_type = CheckType.lint
+    name = "juju-actions"
+    url = "https://juju.is/docs/sdk/charmcraft-analyze#heading--juju-actions"
+    text = "The actions.yaml file is not a valid YAML file."
+
+    # different result constants
+    Result = namedtuple("Result", "ok errors")(ok=OK, errors=ERRORS)
+
+    def run(self, basedir: pathlib.Path) -> str:
+        """Run the proper verifications."""
+        filepath = basedir / "actions.yaml"
+        if not filepath.exists():
+            # it's optional
+            return self.Result.ok
+
+        try:
+            with filepath.open("rt", encoding="utf8") as fh:
+                yaml.safe_load(fh)
+        except Exception:
+            return self.Result.errors
+
+        return self.Result.ok
+
+
+class JujuConfig:
+    """Check that the config.yaml file (if it exists) is valid.
+
+    The file is considered valid if the following checks are true:
+
+    - has an 'options' key
+    - it is a dictionary
+    - each item inside has the mandatory 'type' key
+    """
+
+    check_type = CheckType.lint
+    name = "juju-config"
+    url = "https://juju.is/docs/sdk/charmcraft-analyze#heading--juju-config"
```

Comment:
Slightly rephrased the linked text to fix a typo.

Reply:
thanks!